core/src/main/java/discord4j/core/object/entity/Member.java

```diff
@@ -177,6 +178,20 @@ public String getNicknameMention() {
                 .map(Presence::new);
     }
 
+    /**
+     * Utility to get all permissions on a given user.
+     * 
+     * @return A {@link Mono} where, upon successful completion, emits a PermissionSet containing all effective
+     *         permissions for this member.
+     */
+    public Mono<PermissionSet> getPermissions() {
+        return getGuild().flatMapIterable(Guild::getRoleIds)
+                .filter(role -> getRoleIds().contains(role))
+                .flatMap(id -> getGuild().map(Guild::getClient).flatMap(client -> client.getRoleById(getGuildId(), id)))
```

Comment:
`getClient().getRoleById(getGuildId(), id)` ?

Reply:
Not sure what you are saying

Reply:
Why go through this chain to get to the client when you could just write the above?